code/app.py

```diff
@@ -241,20 +252,21 @@ def stream_without_data(response):
 
 
 def conversation_without_data(request):
-    openai.api_type = "azure"
-    openai.api_base = f"https://{AZURE_OPENAI_RESOURCE}.openai.azure.com/"
-    openai.api_version = "2023-03-15-preview"
-    openai.api_key = AZURE_OPENAI_KEY
+    azure_endpoint = f"https://{AZURE_OPENAI_RESOURCE}.openai.azure.com/"
+    if AZURE_AUTH_TYPE == 'rbac':
+        openai_client = AzureOpenAI(azure_endpoint=azure_endpoint,  api_version=AZURE_OPENAI_API_VERSION, azure_ad_token_provider=AZURE_TOKEN_PROVIDER)
+    else:
+        openai_client = AzureOpenAI(azure_endpoint=azure_endpoint,  api_version=AZURE_OPENAI_API_VERSION, api_key=AZURE_OPENAI_KEY)
 
     request_messages = request.json["messages"]
     messages = [{"role": "system", "content": AZURE_OPENAI_SYSTEM_MESSAGE}]
 
     for message in request_messages:
         messages.append({"role": message["role"], "content": message["content"]})
 
-    response = openai.ChatCompletion.create(
-        engine=AZURE_OPENAI_MODEL,
-        messages=messages,
+    response = openai_client.chat.completions.create(
+        model=AZURE_OPENAI_MODEL,
+        messages = messages,
```

Comment:
Inconsistent spacing around = signs

```suggestion
        messages=messages,
```